stellar/deploy-contract.js

```diff
@@ -139,30 +143,71 @@ async function uploadWasm(filePath, wallet, chain) {
 
 async function upgrade(options, _, chain, contractName) {
     const { wasmPath, yes } = options;
-    const contractAddress = chain.contracts[contractName].address;
+    const uncheckedContractAddress = chain.contracts[contractName].address;
     const wallet = await getWallet(chain, options);
 
     if (prompt(`Proceed with upgrade on ${chain.name}?`, yes)) {
         return;
     }
 
     validateParameters({
-        isNonEmptyString: { contractAddress },
+        isNonEmptyString: { uncheckedContractAddress },
     });
 
+    const contractAddress = Address.fromString(uncheckedContractAddress);
+
     const newWasmHash = await uploadWasm(wasmPath, wallet, chain);
     printInfo('New Wasm hash', serializeValue(newWasmHash));
 
     const operation = Operation.invokeContractFunction({
-        contract: contractAddress,
+        contract: chain.contracts.upgrader.address,
```

Comment:
Include this in the validation above as well in case it's not been deployed/present in the config